Obsidian.API/Plugins/Services/IDiagnoser.cs

```diff
@@ -31,7 +31,7 @@ public interface IDiagnoser : ISecuredService
         /// <param name="createWindow">Indicates whether to start the process in a new window.</param>
         /// <param name="useShell">Indicates whether to use the operating system shell to start the process.</param>
         /// <returns>A new <see cref="IProcess"/> that is associated with the process resource, or null if no process resource is started. </returns>
-        public IProcess StartProcess(string fileName, string arguments = null, bool createWindow = true, bool useShell = false);
+        public IProcess StartProcess(string fileName, string arguments = "", bool createWindow = true, bool useShell = false);
```

Comment:
```suggestion
        public IProcess StartProcess(string fileName, string? arguments = null, bool createWindow = true, bool useShell = false);
```

Could've been made to `string?` since the implementing classes *should've* supported null cases before.

Reply:
Good spot, I did think of this when I made the change but don't know why I thought against it.

Reply:
Having looked at this further the `ProcessStartInfo` which needs the values does NOT support nullable and this can't be changed to support them as it's not our code but Microsofts, I suggest we either follow suit or do a check for nullable and make it an empty string. @Craftplacer

Reply:
I have implemented in the concrete class to default to `string.Empty` if it finds NULL, had a look within the ProcessStartInfo class and it looks like that's what Microsoft is doing themselves.